Packs/AnomaliThreatStreamFeed/Integrations/AnomaliThreatStreamFeed/AnomaliThreatStreamFeed_test.py

```diff
@@ -0,0 +1,978 @@
+    expected_past_time = "2023-08-01T11:00:00.500Z"
+
+    mocker.patch("AnomaliThreatStreamFeed.get_current_utc_time", return_value=mock_now)
+
+    result = get_past_time(minutes_interval)
+    assert result == expected_past_time
+
+
+def test_calculate_score_none_no_confidence_field(mocker):
+    from AnomaliThreatStreamFeed import DBotScoreCalculator
+
+    """
+    Tests calculate_score when the 'confidence' field is missing from the indicator.
+    Verifies that DBotScore.NONE is returned and a debug message is logged.
+
+    Given:
+        - An indicator dictionary without a 'confidence' key.
+    When:
+        - Calling calculate_score.
+    Then:
+        - The function returns Common.DBotScore.NONE.
+        - A debug message indicating confidence not found is logged.
+    """
+    calculator = DBotScoreCalculator()
+    indicator = {"description": "test"}  # No confidence field
+
+    result = calculator.calculate_score(indicator)
+    assert result == Common.DBotScore.NONE
+
+
+DEFAULT_MALICIOUS_THRESHOLD = 65
+DEFAULT_SUSPICIOUS_THRESHOLD = 25
+DEFAULT_BENIGN_THRESHOLD = 0
+DBOT_SCORE_TEST_CASES = [
+    # Test cases for BAD score (confidence > 65)
+    ({"confidence": 71}, Common.DBotScore.BAD),
+    ({"confidence": 100}, Common.DBotScore.BAD),
+    # Test cases for SUSPICIOUS score (confidence > 25 and <= 65)
+    ({"confidence": 51}, Common.DBotScore.SUSPICIOUS),
+    ({"confidence": DEFAULT_MALICIOUS_THRESHOLD}, Common.DBotScore.SUSPICIOUS),  # 65 is not > 65
+    ({"confidence": 60}, Common.DBotScore.SUSPICIOUS),
+    ({"confidence": 26}, Common.DBotScore.SUSPICIOUS),  # Just above suspicious threshold
+    # Test cases for GOOD score (confidence > 0 and <= 25)
+    ({"confidence": 15}, Common.DBotScore.GOOD),
+    ({"confidence": DEFAULT_SUSPICIOUS_THRESHOLD}, Common.DBotScore.GOOD),  # 25 is not > 25
+    ({"confidence": 20}, Common.DBotScore.GOOD),
+    ({"confidence": 1}, Common.DBotScore.GOOD),  # Just above benign threshold
+    # Test cases for NONE score (confidence <= 0)
+    ({"confidence": 0}, Common.DBotScore.NONE),
+    ({"confidence": DEFAULT_BENIGN_THRESHOLD}, Common.DBotScore.NONE),  # 0 is not > 0
+    ({"confidence": -5}, Common.DBotScore.NONE),  # Negative confidence
+    # Test cases for NONE score (missing/invalid confidence)
+    ({"description": "no confidence"}, Common.DBotScore.NONE),  # Missing confidence field
+    ({"confidence": None}, Common.DBotScore.NONE),  # Explicitly None confidence
+    ({"confidence": ""}, Common.DBotScore.NONE),  # Empty string confidence
+]
+
+
+@pytest.mark.parametrize("indicator_input, expected_score", DBOT_SCORE_TEST_CASES)
+def test_calculate_score_various_scenarios(
+    indicator_input: dict[str, Any],
+    expected_score: int,
+):
+    from AnomaliThreatStreamFeed import DBotScoreCalculator
+
+    """
+    Tests calculate_score across various confidence levels and edge cases,
+    including missing or invalid confidence values.
+
```

Comment:
Please explain every test case here, you can also add an ID for each test in the parameterize.